### PR TITLE
Feature/face 140 flow customisation

### PR DESCRIPTION
**Pull request by rfreitas**
the flow can now be customized by adding `steps:[]` to the Onfido object options.

eg:
```javascript
steps: [
  {
     type:'welcome',
     options:{
       title:'Open your new bank account'
     }
   },
   'document',
   'face',
   'complete'
]
```

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -53,8 +54,20 @@
               },
               onComplete: function(event) {
                 // callback for when everything is complete
-              }
-            })
+              },
+              steps: [
```
Comment (hjheath) [inline src/index.html]: Do you think this should go in a separate config file?

File: src/components/app.js [hunk 2]
```diff
@@ -27,7 +27,9 @@ class App extends Component {
   }
 
   render () {
-    const { step, websocketErrorEncountered } = this.props
+    const { websocketErrorEncountered, options } = this.props
```
Comment (hjheath) [inline src/components/app.js]: Can you make a distinction between these options and the per step options

**Comment by hjheath** [comment]
Can you add documentation for this?

**Comment by rfreitas** [comment]
Pull Request on hold:

Missing documentation
Spec has been revised, it should no longer be possible to customize the document selector, it should be a part of the document step;

File: src/components/app.js [hunk 3]
```diff
@@ -19,6 +19,41 @@ import Complete from './Complete'
...
+  const formatStep = typeOrStep => isStep(typeOrStep) ?  typeOrStep : typeToStep(typeOrStep);
+
+  const stepToFormatToComponents = (stepDefaultOptions, step) => stepToComponents(stepDefaultOptions,formatStep(step));
+
+  const shallowFlattenList = list => [].concat(...list);
+
+  return (stepDefaultOptions, steps) => shallowFlattenList(steps.map( step => stepToFormatToComponents(stepDefaultOptions, step)));
+})()
+
+console.log(stepsToComponents);
```
Comment (stephencookdev) [inline src/components/app.js]: Accidental `console.log`?

File: src/components/app.js [hunk 4]
```diff
@@ -19,6 +19,41 @@ import Complete from './Complete'
 
 import styles from '../style/style.css'
 
+const stepsToComponents = ( _ => {
+  //Closure created to encapsulate the set of functionality
```
Comment (stephencookdev) [inline src/components/app.js]: Manually encapsulating logic like this... Maybe this should just be a module?

File: src/index.html (hunk 1 above)
Comment (stephencookdev) [inline src/index.html]: More pressingly, I'm not sure this file should be here at all. Definitely not in its current state.

File: src/components/app.js (hunk 3 above)
Comment (rfreitas) [inline src/components/app.js]: yep.

File: src/components/app.js (hunk 4 above)
Comment (rfreitas) [inline src/components/app.js]: yeah, I was a bit lazy :/ I will put it an module

**Pull request merged**